src/kafka_submitter.rs

```diff
@@ -128,9 +132,18 @@ pub fn share_submitted(state: &KafkaSubmitterState, user_id: &Vec<u8>, user_tag_
 	}));
 }
 
-pub fn weak_block_submitted(state: &KafkaSubmitterState, user_id: &Vec<u8>, user_tag_1: &Vec<u8>, value: u64, header: &BlockHeader, txn: &Vec<Vec<u8>>, _extra_block_data: &Vec<u8>, leading_zeros: u8, required_leading_zeros: u8) {
+pub fn weak_block_submitted(state: &KafkaSubmitterState, user_id: &Vec<u8>, user_tag_1: &Vec<u8>, value: u64, header: &BlockHeader, txn: &Vec<Vec<u8>>, _extra_block_data: &Vec<u8>,
+	leading_zeros: u8, required_leading_zeros: u8, block_hash: &[u8]) {
 	println!("Got valid weak block with value {} from \"{}\" with {} txn from machine identified as \"{}\"", value, String::from_utf8_lossy(user_id), txn.len(), String::from_utf8_lossy(user_tag_1));
-
+	
+	let (block_target, negative, overflow) = utils::nbits_to_target(header.bits);
+	let mut is_good_block = false;
+	if negative || overflow {
+		//TODO: Not sure how to handle this case yet
```

Comment:
Seems like invalid share to me, I'd assume just return :).

Reply:
Sure, thanks